electionleaflets/apps/leaflets/forms.py

```diff
@@ -125,20 +129,23 @@ def get_ballot_data_from_ynr(self, postcode):
 
         :type instance: Leaflet
         """
-        params = {"for_postcode": postcode}
-        start, end = self.get_date_range()
-        params["election_date_range_after"] = start
-        params["election_date_range_before"] = end
-        ynr_helper = YNRAPIHelper()
-        try:
-            resp = ynr_helper.get(
-                "ballots",
-                params=params,
-                json=True,
-            )
-        except requests.RequestException:
-            return []
-        return resp["results"]
+        if "ynr_data" not in self.storage.extra_data:
+            params = {"for_postcode": postcode}
+            start, end = self.get_date_range()
+            params["election_date_range_after"] = start
+            params["election_date_range_before"] = end
+            ynr_helper = YNRAPIHelper()
+            try:
+                resp = ynr_helper.get(
+                    "ballots",
+                    params=params,
+                    json=True,
+                )
+            except requests.RequestException:
+                return []
```

Comment:
I know this is not new in this PR, but it might be helpful to log an error to sentry here before we `return []`
This is scope creep, so feel free to punt it to another PR.